backend/triplestore/rdf_field.py

```diff
@@ -0,0 +1,246 @@
+        field; that is, a subset of the graph that the field is modelling.
+    - `_triples_to_store`: returns a collection of triples that represents the modelled
+        value.
+    
+    An `RDFModel` will use the `get`, `set`, and `clear` endpoints. `set` and `clear` have
+    default implementations based on `_stored_triples` and `_triples_to_store`.
+    Alternatively, you could implement `set` and `clear` directly; in that case, it's not
+    necessary to implement `_stored_triples` and `_triples_to_store`.
+
+    The implementations of the methods listed above should use `self.get_graph()` to
+    fetch the correct graph for reading and writing triples. By default, this will be the
+    graph of the model instance. To use a constant graph, you can pass a `graph` in the
+    initialiser. If the graph is chosen dynamically, you can override the method
+    `get_graph`.
+    '''
+
+    def __init__(self, graph: Optional[Graph] = None, **kwargs):
+        self.graph = graph
+
+    def get_graph(self, instance):
+        '''
+        Return the graph in which to save data.
+
+        By default, this will return the graph passed on on initialisation, if any, or
+        the graph of the model instance.
+
+        You can override this method to choose a graph dynamically.
+        '''
+        return self.graph or instance.graph
+    
+    def get(self, instance):
+        '''
+        Returns the value currently stored in a graph.
+
+        Parameters:
+            instance: the model instance
+        '''
+        raise NotImplementedError()
+
+    def set(self, instance, value) -> None:
+        '''
+        Store a value for the field in a graph
+
+        Parameters:
+            instance: the model instance
+            value: the value of the field on the model instance
+        '''
+        g = self.get_graph(instance)
+        utils.replace_triples(
+            g,
+            self._stored_triples(instance),
+            self._triples_to_store(instance, value)
+        )
+
+    def clear(self, instance) -> None:
+        '''
+        Clear the field's data in a graph for a given instance of the model.
+        
+        This is used as cleanup when deleting the model instance.
+
+        Parameters:
+            instance: the model instance
+        '''
+        g = self.get_graph(instance)
+        for triple in self._stored_triples(instance):
+            g.remove(triple)
+
+    def _stored_triples(self, instance) -> utils.Triples:
+        '''
+        Extract the set of relevant triples from a graph representation
+        '''
+        raise NotImplementedError()
+
+    def _triples_to_store(self, instance, value) -> utils.Triples:
+        '''
+        Triples that should be stored to represent the modelled value
+        '''
+        raise NotImplementedError()
+
+
+class RDFPredicateField(RDFField):
+    '''
+    Models a predicate, i.e. a property + object in the graph. This translates to a
+    boolean value in the model.
```

Comment:
Not very clear to me - perhaps: the `get` method returns a boolean value that is true if the triple exists in the graph, otherwise false? (If that is what you mean)